mkosi/qemu.py

```diff
@@ -1404,13 +1421,16 @@ def add_virtiofs_mount(
                 "-device", f"vhost-user-fs-pci,queue-size=1024,chardev={sock.name},tag={tag}",
             ]  # fmt: skip
 
-            if "fstab.extra" not in credentials:
-                credentials["fstab.extra"] = ""
+            if not (credentials / "fstab.extra").exists():
+                fstab = ""
+            else:
+                fstab = (credentials / "fstab.extra").read_text()
 
-            if credentials["fstab.extra"] and not credentials["fstab.extra"][-1] == "\n":
-                credentials["fstab.extra"] += "\n"
+            if fstab and not fstab[-1] == "\n":
+                fstab += "\n"
```

Comment:
Extra whitespace at the end shouldn't hurt or does it? The logic could be omitted in favour of
```python
    fstab = (credentials / "fstab.extra").read_text() + "\n"
```

Reply:
I'm pretty sure I added this for a reason back then, so I'd rather keep it like this